internal/utils/utils.go

```diff
@@ -293,3 +290,109 @@ func FormatEndpoint(ip string, port int32) string {
 	}
 	return fmt.Sprintf("%s:%d", ip, port)
 }
+
+func DeepHashObject(objToWrite any) string {
+	hasher := fnv.New32a()
+	hasher.Reset()
+	fmt.Fprintf(hasher, "%v", dump.ForHash(objToWrite))
+	return rand.SafeEncodeString(fmt.Sprint(hasher.Sum32()))
+}
```

Comment:
This doesn't return a consistent hash when the order of elements change. Hence we had to sort them earlier.